### DISCUSSION TITLE
Where is the functions computeQpResidual and computeQpjacobian are called?

### DISCUSSION THREAD

**Discussion by Wolke926** [idaholab/moose#25382] [Q&A General] [answered]
I have some trouble understanding the structure of how MOOSE works. In the kernel files, the functions computeQpResidual and computeQpjacobian can be defined but where or in which file are they called?

**Comment by GiudGiud** [comment, marked as answer]
Hello

They are called in the residual and Jacobian loops respectively, see framework/include/loops/ComputeResidualThread.h for example

Guillaume
